chocolatey/NuGet.Client
# (#9) Add new PackageAsync method to ListResource

## Description Of Changes

This commit adds a new PackageAsync method to the ListResource, which can then be used to find out the highest available version of a particular package using one API call.  This can then be consumed within the Chocolatey/choco codebase, when the
UsePackageRepositoryOptimizations option is set, which makes it behave in the same way as the 1.x version of Chocolatey.

## Motivation and Context

When fetching information about a single package, in order to find out the highest available version, NuGet.Client normally has to make multiple calls via the FindPackagesById method, in order to get all package versions, and to then determine which was is highest.

To short circuit this, and make only one call, it is possible to use the Packages() endpoint, providing the exact id to search for, as well as whether it should include prereleases or not.

## Testing

1. Check out this PR, and build the NuGet.Client solution
2. Check out the associated PR into chocolatey/choco - https://github.com/chocolatey/choco/pull/3050
3. Pull the updated NuGet.Client assemblies into the chocolatey/choco project
4. Open Fiddler
5. Run the following command `choco search chocolatey --exact --source chocolatey`
6. Make sure that only calls to the Packages() endpoint are used, and not FindPackagesById
7. Run the following command `choco search chocolatey --exact --source chocolatey --disable-repository-optimizations`
8. Make sure that only calls to the FindPackagesById() endpoint are used, and not Packages()
9. Run the following command `choco info chocolatey --source chocolatey`
10. Make sure that only calls to the Packages() endpoint are used, and not FindPackagesById
11. Run the following command `choco info chocolatey --source chocolatey --disable-repository-optimizations`
12. Make sure that only calls to the FindPackagesById() endpoint are used, and not Packages()
13. Run the following command `choco outdated --source chocolatey`
14. Make sure that only calls to the Packages() endpoint are used, and not FindPackagesById
15. Run the following command `choco outdated --source chocolatey --disable-repository-optimizations`
16. Make sure that only calls to the FindPackagesById() endpoint are used, and not Packages()

### Operating Systems Testing

- Windows 10/11

## Change Types Made


## Change Checklist


## Related Issues

- #9  
- https://app.clickup.com/t/20540031/PROJ-506
